src/sqlfluff/dialects/dialect_oracle.py

```diff
@@ -0,0 +1,81 @@
+"""The Oracle dialect.
+
+This inherits from the ansi dialect.
+"""
+from sqlfluff.core.dialects import load_raw_dialect
+from sqlfluff.core.parser import (
+    BaseSegment,
+    Ref,
+    Sequence,
+    StartsWith,
+    OneOf,
+)
+
+ansi_dialect = load_raw_dialect("ansi")
+oracle_dialect = ansi_dialect.copy_as("oracle")
+
+oracle_dialect.sets("unreserved_keywords").difference_update(["COMMENT"])
+oracle_dialect.sets("reserved_keywords").update(["COMMENT", "ON"])
+
+
+# Adding Oracle specific statements.
+@oracle_dialect.segment(replace=True)
+class StatementSegment(BaseSegment):
+    """A generic segment, to any of its child subsegments."""
+
+    type = "statement"
+
+    parse_grammar = ansi_dialect.get_segment("StatementSegment").parse_grammar.copy(
+        insert=[
+            Ref("CommentStatementSegment"),
+        ],
+    )
+
+    match_grammar = ansi_dialect.get_segment("StatementSegment").match_grammar.copy()
+
+
+@oracle_dialect.segment()
+class CommentStatementSegment(BaseSegment):
+    """A `Comment` statement.
+
+    COMMENT [text]
+    https://docs.oracle.com/cd/B19306_01/server.102/b14200/statements_4009.htm
+    """
+
+    type = "comment_statement"
+
+    match_grammar = StartsWith(Sequence("COMMENT", "ON"))
+
+    parse_grammar = Sequence(
+        "COMMENT",
+        "ON",
+        Sequence(
+            OneOf(
+                Sequence(
+                    OneOf(
+                        "TABLE",
+                        "VIEW",
+                    ),
+                    Ref("TableReferenceSegment"),
+                ),
+                Sequence(
+                    "COLUMN",
+                    Ref("ColumnReferenceSegment"),
+                ),
+                Sequence(
+                    "OPERATOR",
+                    Ref("OperatorReferenceSegment"),
+                ),
+                # Sequence(
+                #     "INDEXTYPE",
+                #     Ref("IndexTypeReferenceSegment"),
+                # ),
+                Sequence(
+                    "MATERIALIZED",
+                    "VIEW",
+                    Ref("MaterializedViewReferenceSegment"),
```

Comment:
I may be wrong here, but I feel we don't have a `MaterializedViewReferenceSegment`.

Referring to Postgres I can see we're using `TableReferenceSegment` like we do for normal views: https://github.com/sqlfluff/sqlfluff/blob/7257d289c08e8a2527117814c1f2d85e8c4f242f/src/sqlfluff/dialects/dialect_postgres.py#L1518